lib/Provider/Command/MessageSend.php

```diff
@@ -0,0 +1,120 @@
+use OCP\Mail\Provider\IAddress;
+use OCP\Mail\Provider\IMessage;
+
+class MessageSend {
+
+	public function __construct(
+		protected IConfig $config,
+		protected ITimeFactory $time,
+		protected AccountService $accountService,
+		protected OutboxService $outboxService,
+		protected AttachmentService $attachmentService
+	) {
+	}
+
+	/**
+	 * performs send operation
+	 *
+	 * @since 4.0.0
+	 *
+	 * @param string $userId system user id
+	 * @param string $serviceId mail account id
+	 * @param IMessage $message mail message object with all required parameters to send a message
+	 * @param array $options array of options reserved for future use
+	 *
+	 * @return LocalMessage
+	 */
+	public function perform(string $userId, string $serviceId, IMessage $message, array $option = []): LocalMessage {
+		// find user mail account details
+		$account = $this->accountService->find($userId, (int)$serviceId);
+		// convert mail provider message to mail app message
+		$localMessage = new LocalMessage();
+		$localMessage->setType($localMessage::TYPE_OUTGOING);
+		$localMessage->setAccountId($account->getId());
+		$localMessage->setSubject($message->getSubject());
+		$localMessage->setBody($message->getBody());
+		// disabled due to issues caused by opening these messages in gui
+		//$localMessage->setEditorBody($message->getBody());
+		$localMessage->setHtml(true);
+		$localMessage->setSendAt($this->time->getTime());
+		
+		// convert all mail provider attachments to local attachments
+		$attachments = [];
+		if (count($message->getAttachments()) > 0) {
+			// iterate attachments and save them
+			foreach ($message->getAttachments() as $entry) {
+				// determine if required parameters are set
+				if (empty($entry->getName()) || empty($entry->getType()) || empty($entry->getContents())) {
+					throw new SendException('Invalid Attachment Parameter: MUST contain values for Name, Type and Contents');
+				}
+				// convert mail provider attachment to mail app attachment
+				$attachments[] = $this->attachmentService->addFileFromString(
+					$userId,
+					$entry->getName(),
+					$entry->getType(),
+					$entry->getContents()
+				)->jsonSerialize();
+			}
+		}
+		// determine if required To address is set
+		if (empty($message->getTo()) || empty($message->getTo()[0]->getAddress())) {
+			throw new SendException('Invalid Message Parameter: MUST contain at least one TO address with a valid address');
+		}
+		// convert recipient addresses
+		$to = $this->convertAddressArray($message->getTo());
+		$cc = $this->convertAddressArray($message->getCc());
+		$bcc = $this->convertAddressArray($message->getBcc());
+		// save message for sending
+		$localMessage = $this->outboxService->saveMessage(
+			$account,
+			$localMessage,
+			$to,
+			$cc,
+			$bcc,
+			$attachments
+		);
+
+		// evaluate if job scheduler is NOT cron, send message right away otherwise let cron job handle it
+		if ($this->config->getAppValue('core', 'backgroundjobs_mode', 'ajax') !== 'cron') {
```

Comment:
@miaulalala wdyt about removing the if and call sendMessage even if the background job is using mail? We don't support scheduling automated emails right now. I don't see a reason to wait for the next background job to push the email out.